### PR TITLE
pin update

### PR DESCRIPTION
**Pull request by lucylq**
Rebasing https://github.com/pytorch/executorch/pull/7235

Changes: https://github.com/pytorch/pytorch/compare/19eff28ff3f19b50da46f5a9ff5f4d4d213806fe..80ca6dd892613fd4f1dee9040b8273ddeadb1c50

### CODE REVIEW

**Review by mergennachin — APPROVED** [review]

**Comment by lucylq** [comment]
Fix related to training IR migration for arm unittest:
https://github.com/pytorch/executorch/pull/7275

**Comment by lucylq** [comment]
QNN failure potentially due to functionalization changes: https://github.com/pytorch/executorch/blob/46acc7169c3109c90ab5ccf6563ace11a7c1e4fb/extension/llm/export/builder.py#L187-L194

File: extension/llm/export/builder.py [hunk 1]
```diff
@@ -193,6 +193,7 @@ def export(self) -> "LLMEdgeManager":
                     dynamic_shapes=dynamic_shape,
                     strict=True,
                 )
+                exported_module.run_decompositions({})
```
Comment (tugsbayasgalan) [inline extension/llm/export/builder.py:196]: You need to re-assign exported_module, it is not an inplace update function.

**Comment by lucylq** [comment]
fix for arm test: https://github.com/pytorch/pytorch/pull/143376

File: extension/llm/export/builder.py [hunk 2]
```diff
@@ -193,6 +193,9 @@ def export(self) -> "LLMEdgeManager":
                     dynamic_shapes=dynamic_shape,
                     strict=True,
                 )
+                print("lfq: pre exported_module ", exported_module.graph_module)
+                exported_module = exported_module.run_decompositions({})
+                print("lfq: post exported_module ", exported_module.graph_module)
```
Comment (lucylq) [inline extension/llm/export/builder.py:198]: @tugsbayasgalan

**Comment by swolchok** [comment]
test-arm-backend-delegation failure appears to be a known problem fixed by @digantdesai in https://github.com/pytorch/pytorch/pull/143376 2 days ago; probably need to roll the version forward?

**Comment by lucylq** [comment]
> test-arm-backend-delegation failure appears to be a known problem fixed by @digantdesai in [pytorch/pytorch#143376](https://github.com/pytorch/pytorch/pull/143376) 14 hours ago; probably need to roll the version forward? (might also need to wait a day; not sure if it's hit a nightly yet or not)

Yeah - looks like it's in the 2024-12-18 nightly, updating: https://github.com/pytorch/pytorch/blob/2ea4b56ec872424e486c4fe2d55da061067a2ed3/torch/fx/passes/infra/partitioner.py#L184

File: extension/llm/export/builder.py [hunk 3]
```diff
@@ -184,15 +184,22 @@ def export(self) -> "LLMEdgeManager":
...
-                exported_module = torch.export.export(
-                    self.model,
-                    self.example_inputs,
-                    self.example_kwarg_inputs,
-                    dynamic_shapes=dynamic_shape,
-                    strict=True,
-                )
+                from unittest.mock import patch
+
+                with patch.object(
```
Comment (tugsbayasgalan) [inline extension/llm/export/builder.py:189]: Can you add a task here before you land it? This is pretty bad hack lol

**Review by tugsbayasgalan — APPROVED** [review]

**Comment by lucylq** [comment]
adding patch for qnn to use old export flow: https://github.com/pytorch/executorch/pull/7370

File: extension/llm/export/builder.py [hunk 4]
```diff
@@ -354,15 +364,25 @@ def export_to_edge(self) -> "LLMEdgeManager":
...
-                self.pre_autograd_graph_module,  # pyre-fixme[6]
-                self.example_inputs,
-                example_kwarg_inputs=self.example_kwarg_inputs,
-                dynamic_shapes=dynamic_shape,
-                edge_constant_methods=self.metadata,
-                edge_compile_config=edge_config,
-                verbose=self.verbose,
-            )
+
+            override_export_behaviour = contextlib.nullcontext()
```
Comment (tugsbayasgalan) [inline extension/llm/export/builder.py:368]: Can you leave a TODO here and mention the issue you filed?

Comment (lucylq) [reply extension/llm/export/builder.py:368]: ah - I left it at the top of the block.

**Comment by lucylq** [comment]
https://github.com/pytorch/pytorch/pull/143423
 
 Pytorch nightly now sets _GLIBCXX_USE_CXX11_ABI by default.

File: third-party/TARGETS [hunk 5]
```diff
@@ -298,7 +298,6 @@ prebuilt_cxx_library(
     name = "libtorch",
     shared_lib = ":libtorch_gen[libtorch]",
     exported_preprocessor_flags = [
-        "-D_GLIBCXX_USE_CXX11_ABI=0",  # `libtorch` is built without CXX11_ABI so any target depends on it need to use the same build config.
```
Comment (huydhn) [inline third-party/TARGETS:301]: Let's see if this works. @atalman Do you know if we have switch libtorch build to CXX11_ABI yet?  I don't see it in https://github.com/pytorch/pytorch/pull/143423 yet

Comment (lucylq) [reply third-party/TARGETS:301]: All the tests are green! Good to land? @huydhn

**Review by huydhn — APPROVED** [review]

**Comment by zingo** [comment]
We see a unittest error on/around this patch that might be caused by it:

 FAILED examples/models/test/test_export.py::ExportTest::test_w2l_export_to_executorch - OSError: /testpath/ExecutorchNightly/executorch/env/lib/python3.10/site-packages/torchaudio/lib/libtorchaudio.so: undefined symbol: _ZNK5torch8autograd4Node4nameEv

**Pull request merged**